src/index.jsx

```diff
@@ -66,20 +66,23 @@ class IntlCurrencyInput extends Component {
 
   updateValues(event) {
     const [value, maskedValue] = this.calculateValues(event.target.value, this.props.config, this.props.currency)
-
-    this.setState({
-      maskedValue,
-    })
-
-    return [value, maskedValue]
+    if (!this.props.max || value < this.props.max) {
+      this.setState({
+        maskedValue,
+      })
+
+      return [value, maskedValue]
+    } else {
+      return [this.normalizeValue(this.state.maskedValue), this.state.maskedValue]
+    }
   }
 
   handleChange(event) {
     event.preventDefault()
 
     const [value, maskedValue] = this.updateValues(event)
 
-    if (this.props.onChange && value && maskedValue) {
+    if (this.props.onChange && maskedValue) {
       this.props.onChange(event, value, maskedValue)
```

Comment:
since zero is falsy, this if statement blocked onChange from triggering when the input was set to zero